src/table/table-body.jsx

```diff
@@ -39,32 +39,23 @@ let TableBody = React.createClass({
   },
 
   getInitialState() {
-    // Determine what rows are 'pre-selected'.
-    let preSelectedRows = [];
-    if (this.props.selectable && this.props.preScanRows) {
-      let index = 0;
-      React.Children.forEach(this.props.children, (child) => {
-        if (React.isValidElement(child)) {
-          if (child.props.selected && (preSelectedRows.length === 0 || this.props.multiSelectable)) {
-            preSelectedRows.push(index);
-          }
-
-          index++;
-        }
-      });
-    }
-
     return {
-      selectedRows: preSelectedRows,
+      selectedRows: this._calculatePreselectedRows(this.props),
     };
   },
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.allRowsSelected && !nextProps.allRowsSelected) {
+    if (nextProps.rowData !== this.props.rowData) {
```

Comment:
This could get tricky. The new table implementation does not use rowData; it relies on children. We'll need to check the children for changes instead of rowData which could get quite expensive if we are doing this check every time the component receives props.